File: benchmarks/tsb/bench_to_dict_oriented.ts
/**
 * Benchmark: toDictOriented (records orient) on 1000x5 DataFrame
 */
import { DataFrame, toDictOriented } from "../../src/index.js";

const ROWS = 1_000;
const WARMUP = 3;
const ITERATIONS = 20;

const df = new DataFrame({
  a: Float64Array.from({ length: ROWS }, (_, i) => i),
  b: Float64Array.from({ length: ROWS }, (_, i) => i * 2),
  c: Float64Array.from({ length: ROWS }, (_, i) => i * 3),
  d: Array.from({ length: ROWS }, (_, i) => `str${i}`),
  e: Float64Array.from({ length: ROWS }, (_, i) => i * 0.5),
});

for (let i = 0; i < WARMUP; i++) {
  toDictOriented(df, "records");
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  toDictOriented(df, "records");
}
const total = performance.now() - start;

console.log(JSON.stringify({ function: "to_dict_oriented", mean_ms: total / ITERATIONS, iterations: ITERATIONS, total_ms: total }));

File: benchmarks/pandas/bench_to_dict_oriented.py
"""Benchmark: DataFrame to_dict(orient='records') on 1000x5 DataFrame"""
import json, time
import numpy as np
import pandas as pd

ROWS = 1_000
WARMUP = 3
ITERATIONS = 20

df = pd.DataFrame({
    "a": np.arange(ROWS, dtype=float),
    "b": np.arange(ROWS, dtype=float) * 2,
    "c": np.arange(ROWS, dtype=float) * 3,
    "d": [f"str{i}" for i in range(ROWS)],
    "e": np.arange(ROWS, dtype=float) * 0.5,
})

for _ in range(WARMUP):
    df.to_dict(orient="records")

start = time.perf_counter()
for _ in range(ITERATIONS):
    df.to_dict(orient="records")
total = (time.perf_counter() - start) * 1000

print(json.dumps({ "function": "to_dict_oriented", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total }))
